packages/create-instance/add-scoped-slots.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+
+import { compileToFunctions } from 'vue-template-compiler'
+import { throwError } from 'shared/util'
+
+export function addScopedSlots (vm: Component, scopedSlots: Object): void {
+  Object.keys(scopedSlots).forEach((key) => {
+    const template = scopedSlots[key].trim()
+    if (template.substr(0, 9) === '<template') {
+      throwError('the scopedSlots option does not support a template tag as the root element.')
+    }
+    const domParser = new window.DOMParser()
+    const document = domParser.parseFromString(template, 'text/html')
```

Comment:
Can you rename this variable to avoid collisions with `window.document`?

Reply:
I changed this.